symtabAPI/src/Object-elf.h

```diff
@@ -148,7 +148,7 @@ class open_statement {
 };
 
 
-class Object : public AObject 
+class Object : public AObject, public boost::basic_lockable_adapter<boost::mutex>
```

Comment:
Could you change this to private inheritance? This represents a has-a rather than an is-a relationship. Generally, I'd prefer composition, but I know `basic_lockable_adapter` is generaly meant to be inherited from.

Reply:
Bah, sorry. I don't know what I was thinking. That adapter has to be publicly inherited from when used with `boost::lock_guard`. I'd like to move to using C++11 constructs for exclusivity, but that's for another day.

Reply:
I've changed it back to public.